src/ucs/sys/sock.c

```diff
@@ -797,6 +798,53 @@ ucs_status_t ucs_sockaddr_get_ifname(int fd, char *ifname_str, size_t max_strlen
     return status;
 }
 
+static ucs_status_t ucs_sockaddr_ifreq(const char *if_name,
+                                       unsigned long request,
+                                       struct sockaddr_in *dest)
+{
+    ucs_status_t status;
+    struct ifreq ifr;
+
+    status = ucs_netif_ioctl(if_name, request, &ifr);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    if ((ifr.ifr_addr.sa_family != AF_INET) ) {
+        ucs_error("%s address is not INET", if_name);
+        return UCS_ERR_INVALID_ADDR;
+    }
```

Comment:
This patch is fixing a bug, which breaks in certain scenarios (OOB behavior is a fatal timeout) - IPv6 support is out-of-scope for this fix, and may be added later.

Specifically, this code was copied verbatim from TCP, and there's no simple way to extend it to IPv6 since `ucs_netif_ioctl()` uses `ifreq` - which does not officially support IPv6 addresses (and `struct lifreq` doesn't seem to be portable). It's not impossible, but it's a task for a separate patch.